### PR TITLE
[REVIEW] Bug cudf & cugraph interoperability issue

### LINKED ISSUE #158: [BUG] Get a warning during cugraph/python install from source
During `python setup.py install` when installing cuGraph from source, I get the following deprecation warning:
```
/conda/envs/rapids/lib/python3.7/site-packages/Cython/Compiler/Main.py:367: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /cugraph/python/cugraph/cugraph.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)
```

@dantegd fyi

### PR DESCRIPTION
**Pull request by seunghwak**
close #158, #168, #170

### CODE REVIEW

**Comment by raydouglass** [comment]
rerun tests

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,10 +7,13 @@
 ## Improvements
 - PR #157 Removed cudatoolkit dependency in setup.py
 - PR #185 Update docs version
+- PR #190 Added a copy optoin in graph creation
 
 ## Bug Fixes
 - PR #169 Disable terminal output in sssp 
 - PR #181 Fixed crash/rmm free error when edge values provided
+- PR #190 Fixed a memory reference counting error between cudf & cugraph
+- PR #190 Fixed a langugae level warning (cython)
```
Comment (ChuckHastings) [inline CHANGELOG.md]: Typo here (language)

File: cpp/include/functions.h [hunk 2]
```diff
@@ -85,7 +85,7 @@ gdf_error gdf_add_adj_list(gdf_graph *graph);
  */
 /* ----------------------------------------------------------------------------*/
 
-gdf_error gdf_add_transpose(gdf_graph *graph);
+gdf_error gdf_add_transposed_adj_list(gdf_graph *graph);
```
Comment (ChuckHastings) [inline cpp/include/functions.h]: Question... would this be better named gdf_add_transposed_from_adj_list (same comment applies to all)?  My first "read" of this function name suggests that it is adding an adj_list... which is in fact the opposite of what it is doing.

Comment (ChuckHastings) [reply cpp/include/functions.h]: By the same token, if we go the route I suggest should gdf_add_edge_list be renamed to gdf_add_edge_list_from_adj_list?

File: CHANGELOG.md (hunk 1 above)
Comment (seunghwak) [reply CHANGELOG.md]: Thanks, I'd better run spelling checker next time :-)

File: cpp/include/functions.h (hunk 2 above)
Comment (seunghwak) [reply cpp/include/functions.h]: Thank you very much for raising this issue. I had similar doubts. So this is related to a bigger API design issue.

First there are naming inconsistencies between python functions and C++/CUDA functions; e.g. add_edge_list in python add_edge_list to the graph from a user provided edge list but gdf_add_edge_list does conversion. And add_edge_list calls C++/CUDA gdf_edge_list_view (not gdf_edge_list).

Second some functions may surprise users; e.g. view_edge_list, view_adj_list, and num_vertices can (implicitly) involve graph conversion.

So we may need to review APIs related to class Graph and C++/CUDA counter part, but this affects multiple parts, I guess we need to solicit options from the entire group. I will submit a separate issue to discuss this.

**Comment by afender** [comment]
@Iroy30 any feedback on this? 
@ChuckHastings do you approve the PR after the changes?

**Comment by ChuckHastings** [comment]
I'm OK tackling the broader issue in #192

**Pull request merged**